tests/test_grid_search.py

```diff
@@ -0,0 +1,243 @@
+    [
+        (
+            SLearner,
+            False,
+            {"base_model": [LinearRegression, LGBMRegressor]},
+            {"base_model": {"LGBMRegressor": {"n_estimators": [1, 2]}}},
+            3,
+        ),
+        (
+            SLearner,
+            True,
+            {"base_model": [LogisticRegression, LGBMClassifier]},
+            {"base_model": {"LGBMClassifier": {"n_estimators": [1, 2]}}},
+            3,
+        ),
+        (
+            TLearner,
+            False,
+            {"variant_outcome_model": [LinearRegression, LGBMRegressor]},
+            {"variant_outcome_model": {"LGBMRegressor": {"n_estimators": [1, 2, 3]}}},
+            4,
+        ),
+        (
+            XLearner,
+            False,
+            {
+                "variant_outcome_model": [LinearRegression],
+                "propensity_model": [LGBMClassifier],
+                "control_effect_model": [LGBMRegressor],
+                "treatment_effect_model": [LGBMRegressor],
+            },
+            {
+                "propensity_model": {"LGBMClassifier": {"n_estimators": [1, 2, 3]}},
+                "control_effect_model": {"LGBMRegressor": {"n_estimators": [1, 2]}},
+                "treatment_effect_model": {"LGBMRegressor": {"n_estimators": [1]}},
+            },
+            6,
+        ),
+        (
+            RLearner,
+            False,
+            {
+                "outcome_model": [LinearRegression],
+                "propensity_model": [LGBMClassifier],
+                "treatment_model": [LGBMRegressor],
+            },
+            {
+                "propensity_model": {"LGBMClassifier": {"n_estimators": [1, 2, 3]}},
+                "treatment_model": {"LGBMRegressor": {"n_estimators": [1, 2, 3]}},
+            },
+            9,
+        ),
+        (
+            DRLearner,
+            False,
+            {
+                "variant_outcome_model": [LinearRegression],
+                "propensity_model": [LGBMClassifier],
+                "treatment_model": [LinearRegression],
+            },
+            {
+                "propensity_model": {"LGBMClassifier": {"n_estimators": [1, 2, 3, 4]}},
+            },
+            4,
+        ),
+    ],
+)
+@pytest.mark.parametrize("n_variants", [2, 5])
+def test_metalearnergridsearch_smoke(
+    metalearner_factory,
+    is_classification,
+    n_variants,
+    base_learner_grid,
+    param_grid,
+    rng,
+    expected_n_configs,
+):
+    metalearner_params = {
+        "is_classification": is_classification,
+        "n_variants": n_variants,
+        "n_folds": 2,
+    }
+    gs = MetaLearnerGridSearch(
+        metalearner_factory=metalearner_factory,
+        metalearner_params=metalearner_params,
+        base_learner_grid=base_learner_grid,
+        param_grid=param_grid,
+    )
+    n_samples = 250
+    n_test_samples = 100
+    X = rng.standard_normal((n_samples, 3))
+    X_test = rng.standard_normal((n_test_samples, 3))
```

Comment:
Wdyt of a `grid_search_data` fixture with both regression and classification outcomes?

Reply:
[acade9e](https://github.com/Quantco/metalearners/pull/9/commits/acade9ea8ab6324c35f2aa977fcc2032eab4b0bc)